src/index.js

```diff
@@ -33,6 +34,17 @@ app.use('/login', loginRouter);
 app.use('/games', gamesRouter);
 app.use('/users', userRouter);
 
-app.listen(port, () => {
+io.on('connection', socket => {
+  console.log('A user connected with socketID: ', socket.id);
+  socket.on('disconnect', () => {
+    console.log('A user disconnected with socketID: ', socket.id);
+  });
+  socket.on('hello', () => {
+    console.log('hello from socketID: ', socket.id);
+  });
+  io.emit('hello');
```

Comment:
프론트엔드에서 서로 다른 유저에 대한 서버의 반응인지 확인할 수 있도록


```suggestion
  io.emit(`hello, user, your socket ID is ${socket.id});
```

와 같이 socket.id 를 클라이언트로 전송하는 것은 어떨까요?

Reply:
좀 맑은 정신으로 보니까 이제 코드를 좀 알겠네요 ㅋㅋ 우선 `io.emit()`을 사용하면 모든 socket들에게 broadcast 하는 것이고, https://socket.io/docs/v4/emitting-events/ 에 따르면 `socket.emit('hello', socket.id)`와 같이 `'hello'`라는 이름의 이벤트로 `socket.id` 변수를 argument로 실어나를 수도 있네요! 반영 하겠습니다

Reply:
@gwonhong 오 ... 넵넵 제가 socket 의 작동원리에 대해 이해가 부족했던 것 같네요. 하하

넵 emit 에 그러면 socket.id 를 실어나르면... 지금으로서는 그러면 특정 유저가 이 연결에 참여했다(?) 의 의미를 전달할 수 있는 것인가요?

Reply:
넵 그런 것 같습니다! 현재 코드(`io.emit()`)로는 socket이 하나라도 접속하면 모든 socket에게 `'hello'`를 broadcast해서, 해당 유저에게만 확인차 socket.id를 보내고자 한다면 `socket.emit('hello', socket.id)`를 사용하면 될 것 같습니다.